### PR TITLE
Remove interop and ci-test features and set interop tests to ignore.

### PR DESCRIPTION
**Pull request by partim**
This PR removes the interop and ci-test features which were used to gate tests that rely on additional software being installed on the test system. This was brittle as it required new features to be added to the ci-test feature or they would not be tested in the CI workflow. Instead, we now simply add the `#[ignore]` attribute to these tests and can use `--all-features` without issue.

### CODE REVIEW

File: src/test/mod.rs [hunk 1]
```diff
@@ -5,10 +5,9 @@
 //!
 //! The actual tests live with their respective modules in a sub-module called
 //! `interop`. Since they require additional software packages to be available
-//! and can be expensive, they are only run if the `"interop"` feature is
-//! given to `cargo test`.
+//! and can be expensive, they are tagged as ignored by default.
 //!
-#![cfg(all(test, feature = "interop"))]
+#![cfg(all(test, feature = "bytes", feature = "std"))]
```
Comment (ximon18) [inline src/test/mod.rs:10]: The line removed from `Cargo.toml` grouped the `bytes` and `ring` features into an `interop` feature yet here you replace `interop` with `bytes` and `std` - is that correct?

Comment (partim) [reply src/test/mod.rs:10]: I believe the code in this module only relies on those two features.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -18,14 +18,13 @@ jobs:
...
-    - run: cargo test --features=ci-test --verbose
+    - run: cargo check --tests --examples --no-default-features
+    - run: cargo test --all-features
     - if: matrix.rust == 'nightly'
       run: |
         cargo +nightly update -Z minimal-versions
-        cargo check --features=ci-test --verbose --all-targets
-        cargo test --features=ci-test
-      name: Check and test with minimal-versions
+        cargo check --all-features --all-targets
```
Comment (ximon18) [inline .github/workflows/ci.yml:29]: Why check more targets here than above?

**Review by ximon18 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (partim) [reply .github/workflows/ci.yml:29]: Fair point. I’ll change the other steps, too.

**Review by ximon18 — APPROVED** [review]

**Pull request merged**